frontend/src/app/main/data/workspace/tokens/remapping_logic.cljs

```diff
@@ -0,0 +1,230 @@
+;; License, v. 2.0. If a copy of the MPL was not distributed with this
+;; file, You can obtain one at http://mozilla.org/MPL/2.0/.
+;;
+;; Copyright (c) KALEIDOS INC
+
+(ns app.main.data.workspace.tokens.remapping-logic
+  "Core logic for token remapping functionality"
+  (:require
+   [app.common.files.changes-builder :as pcb]
+   [app.common.logging :as log]
+   [app.common.types.tokens-lib :as ctob]
+   [app.main.data.changes :as dch]
+   [app.main.data.helpers :as dh]
+   [beicon.v2.core :as rx]
+   [cuerdas.core :as str]
+   [potok.v2.core :as ptk]))
+
+;; Helper Functions
+;; ================
+
+(defn find-token-in-any-set
+  "Find a token by name across all sets in the tokens library"
+  [tokens-lib token-name]
+  (when tokens-lib
+    (let [all-sets (ctob/get-sets tokens-lib)]
+      (some (fn [set]
+              (when-let [token (ctob/get-token-by-name- set token-name)]
+                {:token token :set set}))
+            all-sets))))
+
+;; Token Reference Scanning
+;; ========================
+
+(defn scan-shape-applied-tokens
+  "Scan a shape for applied token references"
+  [shape]
+  (when-let [applied-tokens (:applied-tokens shape)]
+    (for [[attribute token-name] applied-tokens]
+      {:type :applied-token
+       :shape-id (:id shape)
+       :attribute attribute
+       :token-name token-name})))
+
+(defn scan-token-value-references
+  "Scan a token value for references to other tokens (aliases)"
+  [token]
+  (when-let [value (:value token)]
+    (let [value-str (if (string? value) value (str value))]
+      (when (str/includes? value-str "{")
+        ;; Extract token references from value like "{other.token}" or "{color.primary}"
+        (let [references (re-seq #"\{([^}]+)\}" value-str)]
+          (for [[_ token-name] references]
+            {:type :token-alias
+             :source-token-name (:name token)
+             :referenced-token-name token-name}))))))
+
+(defn scan-workspace-token-references
+  "Scan entire workspace for all token references to a specific token"
+  [file-data old-token-name]
+  (let [pages (vals (:pages-index file-data))
+        tokens-lib (:tokens-lib file-data)
+
+        ;; Scan all pages for applied token references
+        applied-refs (mapcat (fn [page]
+                               (let [shapes (vals (:objects page))]
+                                 (mapcat scan-shape-applied-tokens shapes)))
+                             pages)
+
+        ;; Filter for references to the specific token
+        matching-applied (filter #(= (:token-name %) old-token-name) applied-refs)
+
+        ;; Scan tokens library for alias references
+        alias-refs (when tokens-lib
+                     (let [all-tokens (ctob/get-all-tokens tokens-lib)]
+                       (mapcat scan-token-value-references (vals all-tokens))))
+
+        ;; Filter for alias references to the specific token
+        matching-aliases (filter #(= (:referenced-token-name %) old-token-name) alias-refs)]
```

Comment:
Same here.  There may be a lot of tokens, so it's better to add to the list only the ones that match, instead of building a complete list and then another one filtered.